src/sortedDict.jl

```diff
@@ -256,22 +260,55 @@ immutable SDKeyIteration{T <: SDIterableTypesBase}
     base::T
 end
 
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{SortedDict{K,D,Ord}}) = K
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{ExcludeLast{K,D,Ord}}) = K
+eltype{K,D,Ord <: Ordering}(s::SDKeyIteration{IncludeLast{K,D,Ord}}) = K
```

Comment:
While that's true, unspecified type parameters are actually equivalent to `ANY`, in which case, `Array{Complex}` is covariant to `Array{ANY}`:
```julia
julia> Array{Complex} <: Array{Any}
false

julia> Array{Complex} <: Array{ANY}
true
```